source/common/tracing/http_tracer_impl.cc

```diff
@@ -84,10 +84,21 @@ static void addGrpcRequestTags(Span& span, const Http::RequestHeaderMap& headers
 template <class T> static void addGrpcResponseTags(Span& span, const T& headers) {
   addTagIfNotNull(span, Tracing::Tags::get().GrpcStatusCode, headers.GrpcStatus());
   addTagIfNotNull(span, Tracing::Tags::get().GrpcMessage, headers.GrpcMessage());
-  // Set error tag when status is not OK.
+  // Set error tag when Grpc status code represents an error. See
+  // https://github.com/envoyproxy/envoy/issues/18877
   absl::optional<Grpc::Status::GrpcStatus> grpc_status_code = Grpc::Common::getGrpcStatus(headers);
-  if (grpc_status_code && grpc_status_code.value() != Grpc::Status::WellKnownGrpcStatus::Ok) {
-    span.setTag(Tracing::Tags::get().Error, Tracing::Tags::get().True);
+  if (grpc_status_code.has_value()) {
```

Comment:
I think we're going to want to runtime guard this change, as described in CONTRIBUTING.md

Let's also add a comment where the tracing error code is defined to be more clear about what error means in this case (upstream or envoy error, not client error)

Reply:
Runtime guard added, comments about where to find the definition of each grpc status code added.